### DISCUSSION TITLE
Can each service be isolated in a container?

### DISCUSSION THREAD

**Discussion by dlight** [juspay/services-flake#346] [Q&A]
Ok so I didn't find anything about it, but can `services-flake` isolate each service using containers? (Does this even make sense?)

Either by using systemd-nspawn, docker, or by depending on some flake that manages cgroups and namespaces.

**Comment by shivaraj-bh** [comment]
> Ok so I didn't find anything about it, but can services-flake isolate each service using containers?

Can you elaborate on this?

**Reply by dlight** [reply]
I mean, could services-flake create a container for each service, and run inside its own container?

There's some projects to create and run containers using nix flakes, like [extra-containers](https://github.com/erikarvstedt/extra-container) and [flake-containers](https://github.com/adfaure/flake-containers) but they don't come with a list of predefined services like services-flake does.

**Reply by shivaraj-bh** [reply]
> could services-flake create a container for each service, and run inside its own container?

No

At the moment, `services-flake` only exports `processComposeModules` to be used with https://github.com/Platonic-Systems/process-compose-flake. However, in the future we are exploring to export configuration for other backends to consume, see: https://github.com/juspay/services-flake/issues/354
